.github/workflows/main.yml

```diff
@@ -10,13 +10,17 @@ jobs:
       fail-fast: false
       matrix:
         node-version:
+          - 20
+          - 18
+          - 16
           - 14
           - 12
           - 10
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
+          cache: npm
```

Comment:
Also - a matrix this large could do with leveraging an NPM cache to speed this step up. This option was enabled with `actions/setup-node`'s v2 release.